src/python/grpcio_tests/tests/unit/_rpc_part_1_test.py

```diff
@@ -36,6 +37,8 @@
 from tests.unit.framework.common import test_constants
 
 
+@unittest.skipIf(test_common.running_under_gevent(),
+                 "This test is nondeterministic under gevent.")
```

Comment:
optional: The part1 and part2 have large number of cases. Is it possible to have a finer granularity for the skips. E.g., for AsyncIO, the hang/deadlock usually has a pattern, like only happen to a type of RPC, or only happen to non-OK issues.

But if it's too time-costly to check, I'm okay to keep it this way.

Reply:
I attempted to bisect and found that 90+% of the individual test cases deterministically deadlock. I do intend to make these tests work with the gevent reimplementation, so I'm not giving up on these yet.